lib/ukconsole/Makefile.uk

```diff
@@ -0,0 +1,10 @@
+$(eval $(call addlib_s,libukconsole,$(CONFIG_LIBUKCONSOLE)))
+
+CINCLUDES-$(CONFIG_LIBUKCONSOLE) += -I$(LIBUKCONSOLE_BASE)/include
+CXXINCLUDES-$(CONFIG_LIBUKCONSOLE) += -I$(LIBUKCONSOLE_BASE)/include
+
+LIBUKCONSOLE_SRCS-y += $(LIBUKCONSOLE_BASE)/console.c|isr
```

Comment:
The console functions may run very early in the boot sequence. At that point , we don't want to assume that the FPU, SIMD, etc. are properly initialized. Compiling ISR safe keeps to compiler from using SIMD instructions for example. You, @michpappas , said before that in practice, these parts of the CPU are already initialize when the console functions run. Maybe compiling them ISR safe is premature future-proofing. @skuenzer What do you think

Reply:
Indeed. Right now everything is initialized before early init. Since initializing SIMD in `lcpu_init()` is something TBD I would save such changes to that time. @skuenzer is that ok with you?

Reply:
This is a very good point. So far, we used the `|isr` approach for early boot code but you are right @michpappas, actually not needed. I think it would make sense if we deploy the following convention/assumption: Any CPU feature requested by compiler (via selected target CPU under arch/) is initialized in early assembly code, if missing or fails, we stop boot with an very short and static error message, like `"Unsupported CPU"`. Any additional feature, we should initialize later during boot and provide an option if this should happen: Use case is `elfloader` where we run binary code that could be compiled differently than the native Unikraft kernel.
Overall, if we all agree, we should aim this structure with the new `lib/ukboot` concept, and then `|isr` is not needed in fact.
For now, personally, I am okay with two options:
- We document (probably) in the assembler code that we have to initialize compiler-requested CPU features before jumping into C code; and we remove `|isr` here.
- We keep `|isr` and document here a `TODO/FIXME` that says that `|isr` can be removed as soon we have the just mentioned scheme officially in place.

Reply:
That's fine, in that case we'll drop the `isr` flag and go with a comment. I will update #1472 that introduces the earlytab as that seems a more appropriate place.